src/unfold.js

```diff
@@ -3,35 +3,37 @@ var _curry2 = require('./internal/_curry2');
 
 /**
  * Builds a list from a seed value. Accepts an iterator function, which returns
- * either false to stop iteration or an array of length 2 containing the value
- * to add to the resulting list and the seed to be used in the next call to the
- * iterator function.
+ * either an empty array to stop iteration, or a singleton array containing an
+ * array of length 2, with the value that's added to the resulting list and the
+ * seed to be used in the next call of the iterator function.
  *
  * The iterator function receives one argument: *(seed)*.
  *
  * @func
  * @memberOf R
  * @since v0.10.0
  * @category List
- * @sig (a -> [b]) -> * -> [b]
+ * @sig (b -> [(a, b)]) -> b -> [a]
```

Comment:
I wouldn't mind any solution, but since we're asking ourselves this question, I would propose to rely on a boolean value for these checks. It will make the code cleaner, and it will be easier to understand for newcomers. Otherwise I'm up for the use of a generic Maybe exposed through Ramda's API. We could even use a [Symbol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol) for this.